src/components/CreateVmWizard/helpers.js

```diff
@@ -0,0 +1,101 @@
+import { createTemplateList } from '_/components/utils'
+import timezones from '_/components/utils/timezones.json'
+import { isWindows } from '_/helpers'
+
+const handleClusterIdChange = (clusterId, blankTemplateId, { templates, clusters }) => {
+  const changes = {}
+  if (clusterId === undefined) {
+    changes.dataCenterId = undefined
+    changes.clusterId = undefined
+    changes.provisionSource = 'template'
+    changes.isoImage = undefined
+    changes.templateId = undefined
+  } else {
+    const templateList = createTemplateList(templates, clusterId)
+
+    changes.dataCenterId = clusters.getIn([clusterId, 'dataCenterId'])
+    changes.clusterId = clusterId
+    changes.provisionSource =
+      (templateList.length === 1 && templateList[0].id === blankTemplateId)
+        ? 'iso'
+        : 'template'
+    changes.isoImage = undefined
+    changes.templateId = templateList.length === 1 ? blankTemplateId : undefined
+  }
+  return changes
+}
+
+const handleProvisionSourceChange = (provisionSource, defaultValues) => {
+  const changes = {}
+  changes.provisionSource = provisionSource
+  changes.isoImage = undefined
+  changes.templateId = undefined
+  changes.operatingSystemId = defaultValues.operatingSystemId
+  changes.memory = defaultValues.memory
+  changes.cpus = defaultValues.cpus
+  changes.optimizedFor = defaultValues.optimizedFor
+  changes.cloudInitEnabled = defaultValues.initEnabled
+
+  // when changing Provision type to ISO, set the default time zone according to the OS
+  if (changes.provisionSource === 'iso') {
+    changes.timeZone = checkTimeZone(changes.operatingSystemId, changes.templateId)
+  }
+  return changes
+}
+
+const handleTemplateIdChange = (templateId, optimizedFor, { templates, operatingSystems, defaultGeneralTimezone, defaultWindowsTimezone }) => {
```

Comment:
Do we have to get the `defaultGeneralTimezone` and `defaultWindowsTimezone` like this, as function parameters? Those values are always the same. It would be nice if it was possible and if we had those values available in the whole helper, so we could simplify the code (not only) here.

Reply:
Both values originate from `props` and in theory can change with every render.
We could use some tricks to make the param list look shorted:
1.  continue destructuring in  the param list but pack unused params using rest pattern i.e. ` {templates, ...otherProps}`
     Then we can restore the object using spread operator if needed i.e.  `const restoredProps = { templates, ...otherProps}`
2.  don't do destructuring in the param list. Move it to the body and take only what you need

Reply:
Both approaches assume that we don't care about getting more properties then needed - destructuring and restoring has a (nice) side effect of filtering out alien properties.